src/batch/src/task/task_execution.rs

```diff
@@ -43,6 +46,46 @@ use crate::task::BatchTaskContext;
 // Now we will only at most have 2 status for each status channel. Running -> Failed or Finished.
 const TASK_STATUS_BUFFER_SIZE: usize = 2;
 
+/// A special version for batch allocation stat, passed in another task context to report task mem
+/// usage 0 bytes at the end.
+pub async fn allocation_stat_for_batch<Fut, T, F, C>(
+    future: Fut,
+    interval: Duration,
+    mut report: F,
```

Comment:
We don't need this? I think the report here is just calling `BatchTaskContext`?

Reply:
Yes.. But if we want to report 0 bytes in the end, we need to pass another context ref (the previous one is moved to closure)